### DISCUSSION TITLE
How can I get md5 of multipart upload file ?

### DISCUSSION THREAD

**Discussion by SinnoSong** [minio/minio#17893] [Q&A] [answered]
I use multipart upload to upload a file to Minio, but the eTag is md5 of latest part. How can I get the md5 of this file?

**Comment by klauspost** [comment, marked as answer]
The S3 API does not provide MD5 of multipart uploads, not if it is stored encrypted.

**Reply by SinnoSong** [reply]
So I must read this file to get md5 by myself?

**Reply by klauspost** [reply]
Correct.
